osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -109,15 +109,23 @@ protected override void UpdateAfterChildren()
 
             comboDisplay.X = Catcher.X;
 
-            if (!lastHyperDashState && Catcher.HyperDashing && Time.Elapsed > 0)
-                catcherTrails.DisplayHyperDashAfterImage(Catcher.CurrentState, Catcher.X, Catcher.BodyScale);
+            if (Time.Elapsed <= 0)
+            {
+                // This is probably a wrong value, but currently the true value is not recorded.
+                // Setting `true` will prevent generation of false-positive after-images (with more false-positives).
+                lastHyperDashState = true;
```

Comment:
Can you explain how this false-positive generation happens? I'd like to think the negative rate/elapsed check should suffice to ensure that no extra entries are added during a rewind...?

Reply:
The comment is wrong and

```suggestion
                // Setting `true` will prevent generation of false-positive after-images (with more false-negatives).
                lastHyperDashState = true;
```

And, without `lastHyperDashState = true`, false positives happen.
When a hyper-fruit is caught when the catcher was already hyper-dashing, no afterimage should be displayed. But `lastHyperDashState` could be false when a replay is rewound to the time between the first hyper-fruit and the second hyper-fruit.

Reply:
Ok, I see, that explains it, thanks.

Mostly spitballing here, but could this instead be fixed by maintaining a "last trail spawned" time, and never spawning trails if the current time is less than that last trail spawn time? The rationale in my head is that if an afterimage entry has been spawned for a hitobject already prior to a rewind, there is no point in doing any more forward processing again *after* the rewind (at least until the gameplay catches up to where it was previously).

Reply:
That works for a replay, but all other codes don't assume the play is not changing after a rewind.
Specifically, the replay can change after a rewind in the current editor (consider: rewind -> edit -> play).
Also, there is a future possibility of something like a quicker quick-retry, I imagine.
Note that it is specifying `RemoveRewoundEntry` to remove entries when rewound.

Reply:
Ok, I guess that doesn't universally work. Probably fine to let be for now, and eventually the info about the hyperdash trigger time should be stored to the judgements, probably. mania does similar things with similar effects from what I recall.